server/lib/scheduler/Schedule.ts

```diff
@@ -0,0 +1,142 @@
+import { map } from "asyncro";
+import * as _ from "lodash";
+import { UserSchedule } from "./UserSchedule";
+
+class Schedule {
+  private cacheForWhenTreeLastWorkedOn = {};
+  private rootParentCache;
+  private schedule = new Map;
+  private timeLimit;
+
+  public constructor({ rootParentCache, timeLimit }) {
+    this.rootParentCache = rootParentCache;
+    this.timeLimit = timeLimit;
+  }
+
+  private doesUserHaveASchedule(userId) {
+    return this.schedule.has(userId);
+  }
+
+  private createUserSchedule(userId) {
+    this.schedule.set(userId, new UserSchedule({ timeLimit: this.timeLimit, userId }));
+  }
```

Comment:
Would it be ok for this to be called multiple times with one userId? If not, I suggest adding a guard, then call this from the ``assignWorkspaceToUser`` fn. Maybe rename to `createUserScheduleIfNotCreated`